src/yarp.c

```diff
@@ -11218,42 +11235,109 @@ parse_expression_prefix(yp_parser_t *parser, yp_binding_power_t binding_power) {
+                            &parser->current);
+
                 } else {
-                    yp_interpolated_string_node_append((yp_interpolated_string_node_t *) node, part);
+                    node = (yp_node_t *)yp_string_node_create(
+                            parser,
+                            &parser->previous,
+                            &((yp_token_t) { .type = YP_TOKEN_STRING_CONTENT, .start = parser->previous.end, .end = parser->current.start }),
+                            &parser->current);
                 }
+                node->location = (yp_location_t) {
+                    .start = parser->previous.start,
+                        .end = parser->previous.end
+                };
+                lex_state_set(parser, YP_LEX_STATE_END);
+                expect(parser, YP_TOKEN_HEREDOC_END, YP_ERR_HEREDOC_TERM);
+                return node;
             }
 
-            lex_state_set(parser, YP_LEX_STATE_END);
-            expect(parser, YP_TOKEN_HEREDOC_END, YP_ERR_HEREDOC_TERM);
+            yp_token_t opening_token = parser->previous;
 
-            if (quote == YP_HEREDOC_QUOTE_BACKTICK) {
-                assert(YP_NODE_TYPE_P(node, YP_INTERPOLATED_X_STRING_NODE));
-                yp_interpolated_xstring_node_closing_set(((yp_interpolated_x_string_node_t *) node), &parser->previous);
-                node->location = ((yp_interpolated_x_string_node_t *) node)->opening_loc;
-            } else {
-                assert(YP_NODE_TYPE_P(node, YP_INTERPOLATED_STRING_NODE));
-                yp_interpolated_string_node_closing_set((yp_interpolated_string_node_t *) node, &parser->previous);
-                node->location = ((yp_interpolated_string_node_t *) node)->opening_loc;
+            yp_node_t *part = parse_string_part(parser);
+
+            if (part == NULL) {
+                // We couldn't parse anything, so return a missing node
+                return (yp_node_t *) yp_missing_node_create(parser, parser->previous.start, parser->previous.end);
             }
 
-            // If this is a heredoc that is indented with a ~, then we need to dedent
-            // each line by the common leading whitespace.
-            if (indent == YP_HEREDOC_INDENT_TILDE) {
-                parse_heredoc_dedent(parser, node, quote);
+            if (YP_NODE_TYPE_P(part, YP_STRING_NODE) && match_any_type_p(parser, 2, YP_TOKEN_HEREDOC_END, YP_TOKEN_EOF)) {
+                // We only have a single string, so we can return it
+                yp_string_node_t *str_part = (yp_string_node_t *)part;
+                (str_part)->opening_loc = (yp_location_t) {
+                    .start = opening_token.start,
+                    .end = opening_token.end
+                };
+                (str_part)->closing_loc = (yp_location_t) {
+                    .start = parser->current.start,
+                    .end = parser->current.end
+                };
+                (str_part)->base.location = (yp_location_t) {
```

Comment:
I think you can simplify this to `str_part->base.location = str_part->opening_loc` since you just set it.